File: src/main/java/com/tabnine/logging/SentryAppender.java
package com.tabnine.logging;

import io.sentry.Sentry;
import org.apache.log4j.AppenderSkeleton;
import org.apache.log4j.spi.LoggingEvent;

public class SentryAppender extends AppenderSkeleton {

    @Override
    protected void append(LoggingEvent loggingEvent) {
        if (loggingEvent.getThrowableInformation() != null) {
            Sentry.captureException(loggingEvent.getThrowableInformation().getThrowable());
        }
    }

    @Override
    public void close() {

    }

    @Override
    public boolean requiresLayout() {
        return false;
    }
}

File: src/main/java/com/tabnine/logging/SentryAppender.kt
package com.tabnine.logging

import io.sentry.Sentry
import org.apache.log4j.AppenderSkeleton
import org.apache.log4j.spi.LoggingEvent

class SentryAppender : AppenderSkeleton() {
    override fun append(loggingEvent: LoggingEvent) {
        if (loggingEvent.throwableInformation != null) {
            Sentry.captureException(loggingEvent.throwableInformation.throwable)
        }
    }

    override fun close() {}
    override fun requiresLayout(): Boolean {
        return false
    }
}
